tests/test_api/test_categories.py

```diff
@@ -1,45 +1,60 @@
 import pytest
 import json
 
+from . import TestBase
+
 from sourcemash.models import Feed
 
 def check_valid_header_type(headers):
     assert headers['Content-Type'] == 'application/json'
 
-class TestCategoryListAPI:
+class TestCategoryListAllAPI:
     def test_get_categories_present(self, test_client, itemWithCategory):
-        r = test_client.get('/api/categories')
+        r = test_client.get('/api/categories/all')
         check_valid_header_type(r.headers)
         assert r.status_code == 200
 
         data = json.loads(r.data)
-        assert data['categories'][0]['category'] == itemWithCategory.category_1 or \
-                data['categories'][0]['category'] == itemWithCategory.category_2
-        assert data['categories'][1]['category'] == itemWithCategory.category_1 or \
-                data['categories'][1]['category'] == itemWithCategory.category_2
+        assert set([data['categories'][0]['category'], data['categories'][1]['category']]) == \
+                set([itemWithCategory.category_1, itemWithCategory.category_2])
 
-class TestCategoryItemListAPI:
 
-    def test_get_category_items_present(self, test_client, itemWithCategory):
-        r = test_client.get('/api/categories/' + itemWithCategory.category_1)
+class TestCategoryListAPI(TestBase):
+    def test_get_user_categories_unsubscribed(self, test_client, user):
+        self.login(test_client, user.email, user.password)
```

Comment:
something like TestBase.login isn't required to use this? How does self.login even work here?